File: src/spice2x/api/resources/python/spiceapi/touch.py
from .connection import Connection
from .request import Request


def touch_read(con: Connection):
    res = con.request(Request("touch", "read"))
    return res.get_data()


def touch_write(con: Connection, touch_points):
    req = Request("touch", "write")
    for state in touch_points:
        req.add_param(state)
    con.request(req)


def touch_write_reset(con: Connection, touch_ids):
    req = Request("touch", "write_reset")
    for touch_id in touch_ids:
        req.add_param(touch_id)
    con.request(req)

File: src/spice2x/api/resources/dart/spiceapi-websocket/src/wrappers/touch.dart
part of spiceapi;

class TouchState {
  int id;
  int x, y;

  TouchState(this.id, this.x, this.y);
}

Future<List<TouchState>> touchRead(Connection con) {
  var req = Request("touch", "read");
  return con.request(req).then((res) {

    // build states list
    List<TouchState> states = [];
    for (List state in res.getData()) {
      states.add(TouchState(
        state[0],
        state[1],
        state[2],
      ));
    }

    // return it
    return states;
  });
}

Future<void> touchWrite(Connection con, List<TouchState> states) {
  var req = Request("touch", "write");

  // add params
  for (var state in states) {
    var obj = [
      state.id,
      state.x,
      state.y
    ];
    req.addParam(obj);
  }

  return con.request(req);
}

Future<void> touchWriteReset(Connection con, List<int> touchIDs) {
  var req = Request("touch", "write_reset");

  // add params
  for (var id in touchIDs)
    req.addParam(id);

  return con.request(req);
}
